src/wp-includes/formatting.php

```diff
@@ -4907,7 +4907,7 @@ function sanitize_option( $option, $value ) {
 			break;
 	}
 
-	if ( ! empty( $error ) ) {
+	if ( null !== $error ) {
```

Comment:
Differentiating between `null` and an empty string makes sense to me.

However, to avoid passing an empty string to `add_settings_error()`, we should probably add a fallback message like this:
```
if ( '' === $error && is_wp_error( $value ) ) {
	$error = sprintf(
		/* translators: 1: Option name, 2: Error code. */
		__( 'Could not sanitize the %1$s option. Error code: %2$s' ),
		$option,
		$value->get_error_code()
	);
}
```
That should help with debugging :)